### PR TITLE
:recycle: Replace dMonster usage on DoAttack

### PR DESCRIPTION
**Pull request by canassa**

### CODE REVIEW

File: Source/player.cpp [hunk 1]
```diff
@@ -1055,10 +1052,10 @@ bool DoAttack(Player &player)
 			}
 		}
 
-		if (dMonster[dx][dy] != 0) {
-			didhit = PlrHitMonst(player, abs(dMonster[dx][dy]) - 1);
-		} else if (dPlayer[dx][dy] != 0 && !player.friendlyMode) {
-			didhit = PlrHitPlr(player, Players[abs(dPlayer[dx][dy]) - 1]);
+		if (monster != nullptr) {
+			didhit = PlrHitMonst(player, *monster);
+		} else if (monster != nullptr && !player.friendlyMode) {
```
Comment (ephphatha) [inline Source/player.cpp:1057]: monster  != nullptr is the wrong check here, with that change this case will never be true.
```suggestion
		} else if (IsPlayerAtPosition(position) && !player.friendlyMode) {
```

Comment (canassa) [reply Source/player.cpp]: Thanks!

Comment (AJenbo) [reply Source/player.cpp]: You will also have to implement `IsPlayerAtPosition`

Comment (canassa) [reply Source/player.cpp]: I replaced it with `PlayerAtPosition`

File: Source/main.cpp [hunk 2]
```diff
@@ -1,3 +1,6 @@
+#if (defined(_WIN64) || defined(_WIN32)) && defined(__GNUC__)
```
Comment (qndel) [inline Source/main.cpp:1]: what's this

Comment (canassa) [reply Source/main.cpp:1]: Rebase error with master, let me fix

File: Source/player.cpp [hunk 3]
```diff
@@ -1055,10 +1052,10 @@ bool DoAttack(Player &player)
 			}
 		}
 
-		if (dMonster[dx][dy] != 0) {
-			didhit = PlrHitMonst(player, abs(dMonster[dx][dy]) - 1);
-		} else if (dPlayer[dx][dy] != 0 && !player.friendlyMode) {
-			didhit = PlrHitPlr(player, Players[abs(dPlayer[dx][dy]) - 1]);
+		if (monster != nullptr) {
+			didhit = PlrHitMonst(player, *monster);
+		} else if (PlayerAtPosition(position) != nullptr && !player.friendlyMode) {
```
Comment (ephphatha) [inline Source/player.cpp:1057]: There's no nice way to structure this to avoid calling `PlayerAtPosition` twice(or doing it unnecessarily before the if statements :/). Would be worth reordering the conditions so the PVP flag is checked first but I'd do that in a separate PR and include things like the calls to `CanTalkToMonst` in the cleave section.

**Review by ephphatha — APPROVED** [review]

File: Source/player.cpp (hunk 3 above)
Comment (canassa) [reply Source/player.cpp:1057]: There's the "If statement with initializer" from C++ 17 but I heard that that feature is not very popular 😁

Comment (ephphatha) [reply Source/player.cpp:1057]: yeah, I'd rate that a "not-nice" way of doing it :D
